### ISSUE TITLE
Refactor Brave Ads anonymity set rules to be dynamic

### ISSUE THREAD

**Issue by moritzhaller** [brave/brave-browser#12437]
Implement dynamic Anonymity Set Rules for Brave Ads

**Comment by ShivanKaul** [comment]
STAR should address this, closing

**Comment by tmancey** [comment]
Reopening issue as we dont think this maps 1:1 cc @moritzhaller to confirm

**Comment by moritzhaller** [comment]
@ShivanKaul I'm not sure STAR would be suitable for confirmations as we need accurate accounting for ad events. It might be useful to have more granular (and non necessary) meta data behind STAR, but I think we need to discuss what is essential vs. what can be send with STAR. Does that make sense?

**Comment by ShivanKaul** [comment]
Yeah, @evq and I have been discussing this a bit, perhaps nested STAR could work here. I might be missing context, what was the alternative you were considering here @moritzhaller?

**Comment by moritzhaller** [comment]
@ShivanKaul coming back to this now, STAR should in fact work for the kind of data we would have protected with the dynamic k-anon heuristics in question. We will always send the ad event, and k-anon considerations only concern more detailed meta data. So protecting this meta data with STAR should be strictly better than using heuristics - in theory. So if (nested) STAR is actually viable we could rename the ticket

**Comment by tmancey** [comment]
@moritzhaller could you please rename the ticket, thanks
